packages/block-editor/src/components/media-upload/index.native.js

```diff
@@ -0,0 +1,98 @@
+/**
```

Comment:
This is a new component to be used by both video and image blocks also by MediaUpload component. It is actually not a new code, just extracted from image block.

This component is responsible for displaying media options picker to user. It is designed similar to what web has, it passes a function 'open' via its render method and the rendered component decides when to call that function. 

Different thing is web doesn't need to pass `getMediaOptions` but we do, the renderer also needs to embed `getMediaOptions` call to an appropriate place. I tried wrapping this component with a `View` and calling `getMediaOptions` inside, but that messed up the layout of edit button. 🤷‍♀️